x-pack/plugins/uptime/public/pages/monitor.tsx

```diff
@@ -15,6 +15,20 @@ import { useMonitorId, useUptimeTelemetry, UptimePage } from '../hooks';
 import { MonitorCharts } from '../components/monitor';
 import { MonitorStatusDetails, PingList } from '../components/monitor';
 import { getDynamicSettings } from '../state/actions/dynamic_settings';
+import { Ping } from '../../common/runtime_types/ping';
+
+const isAutogeneratedId = (id: string) => {
+  return /^auto-(icmp|http|tcp)-OX[A-F0-9]{16}-[a-f0-9]{16}/.test(id);
+};
+
+// For auto Gen Id we will display url since ID looks too ugly in the UI
+// We will plan to deprecate it after 8.0.0 once is required in heartbeat config
+const getIdToDisplay = (monId: string, selectedMonitor: Ping | null) => {
+  if (isAutogeneratedId(monId)) {
+    return selectedMonitor?.url?.full || monId;
+  }
+  return monId;
```

Comment:
```suggestion
// For monitors with no explicit ID, we display the URL instead of the
// auto-generated ID because it is difficult to derive meaning from a
// generated id like `auto-http-0X8D6082B94BBE3B8A`.
//
// We may deprecate this behavior in the next major release, because
// the heartbeat config will require an explicit ID.
const getIdToDisplay = (monitorId: string, selectedMonitor: Ping | null) => {
  if (isAutogeneratedId(monitorId)) {
    return selectedMonitor?.url?.full || monitorId;
  }
  return monitorId;
```

I think we should define an issue to track this addition for 8.0 if we want to do it (alternatively we could remove the second line of comments); in my experience if we say we will do something in comments but don't track it outside of the code, we tend to forget about the promise.